test/e2e/service.go

```diff
@@ -112,25 +116,25 @@ var _ = Describe("Services", func() {
 		addEndpointPodOrFail(c, ns, name1, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name1)
 
-		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}})
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]string{name1: {portname1}})
```

Comment:
The endpoint might have a different IP than the container IP and a different port than the service port. The change above uses named port in order to get the matching right in the test between the service and the endpoint b/c this is verified in the test. And because the port is not suitable anymore for the matching, we need these names as an alternative.

It's important to note that this change is only necessary for these very tests which check the existence of correct endpoints. Of course other services in different tests don't need named ports.

I will add proper comments.